server.py

```diff
@@ -33,17 +33,10 @@ def download(number):
 
 @app.route('/upload', method = 'POST')
 def file_upload():
-    upload = request.files.get('upload')
-    name, ext = os.path.splitext(upload.filename)
-
-    save_path = "./files/user_outputs/"
-    upload.save(save_path)  # appends upload.filename automatically
-    
-    userpath = "./files/user_outputs/" + name + '.' + ext
-
-    with open('./files/expected_outputs/expected_output.txt','r') as f2 , open(userpath,'r') as f1:
-        ans = True
-        ans = f1.read().strip() == f2.read().strip()
+    uploaded = request.files.get('upload').file.read() #uploaded outputs by user
+    expected.strip()
+    uploaded.strip()
+    ans = (uploaded==expected)
 
     if ans==False:
```

Comment:
Comparison with Booleans is not recommended.
```suggestion
    if not ans:
```